app/src/test/java/com/x8bit/bitwarden/ui/auth/feature/accountsetup/SetupUnlockScreenTest.kt

```diff
@@ -0,0 +1,478 @@
+package com.x8bit.bitwarden.ui.auth.feature.accountsetup
+
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.assertIsEnabled
+import androidx.compose.ui.test.assertIsOff
+import androidx.compose.ui.test.assertIsOn
+import androidx.compose.ui.test.filterToOne
+import androidx.compose.ui.test.hasAnyAncestor
+import androidx.compose.ui.test.isDialog
+import androidx.compose.ui.test.onAllNodesWithText
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performScrollTo
+import androidx.compose.ui.test.performTextInput
+import com.x8bit.bitwarden.data.platform.repository.util.bufferedMutableSharedFlow
+import com.x8bit.bitwarden.ui.platform.base.BaseComposeTest
+import com.x8bit.bitwarden.ui.platform.components.toggle.UnlockWithPinState
+import com.x8bit.bitwarden.ui.platform.manager.biometrics.BiometricsManager
+import com.x8bit.bitwarden.ui.util.assertNoDialogExists
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.runs
+import io.mockk.verify
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.update
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.robolectric.annotation.Config
+
+class SetupUnlockScreenTest : BaseComposeTest() {
+
+    private var onNavigateToSetupAutofillCalled = false
+
+    private val biometricsManager: BiometricsManager = mockk {
```

Comment:
💭 Could we add a test where if this is false we confirm the switch is not displayed?

Reply:
I did try that bit changing that value will not force the test to recompose the UI, so it never changes:

So this always fails on `assertDoesNotExist()`
```kotlin
    @Test
    fun `biometrics toggle should update according to biometrics manager`() {
        every { biometricsManager.isBiometricsSupported } returns true
        composeTestRule
            .onNodeWithText(text = "Unlock with Biometrics")
            .assertExists()

        every { biometricsManager.isBiometricsSupported } returns false
        composeTestRule
            .onNodeWithText(text = "Unlock with Biometrics")
            .assertDoesNotExist()
    }
```

Reply:
yeah you would have to do a separate `composeTestRule.setContent{}` with separate mocking. I believe it would overwrite the one in the `@Before`. But if you don't feel like its needed I am fine since we probably don't have this one off elsewhere?